tests/core/test_data.py

```diff
@@ -707,3 +708,60 @@ def test_save_to_netcdf(mocker, caplog, fixture_data, out_path, raises, exp_log)
         Path(out_path).unlink()
 
     log_check(caplog, exp_log)
+
+
+def test_Data_replace_from_dict(dummy_climate_data):
+    """Test reading from dictionary."""
```

Comment:
I'm not sure this is the correct way of testing this, could you please check?

Reply:
LGTM